storage_controller/src/service.rs

```diff
@@ -905,9 +910,7 @@ impl Service {
     async fn spawn_heartbeat_driver(&self) {
         self.startup_complete.clone().wait().await;
 
-        const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
-
-        let mut interval = tokio::time::interval(HEARTBEAT_INTERVAL);
+        let mut interval = tokio::time::interval(self.config.heartbeat_interval);
```

Comment:
this is ready right away isn't it?

I think it is: https://docs.rs/tokio/latest/tokio/time/fn.interval.html

There's also the default burst behaviour, which might come in to play here, at least @problame has fixed all of my use of interval to be just "sleep for duration after one attempt".

Perhaps it doesn't need to be fixed this time around.

Reply:
Storage controller starts first and pageservers second. Pagserver's re-attach and then we wait for a heartbeat response to mark the node active. The re-attach can land at any point during the interval, so I don't think it matters if it's ready immediately.

Maybe I misunderstood the suggestion.

Reply:
Resolving to merge if tests are happy. Let's discuss it separately